browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -2300,11 +2300,21 @@ L.CanvasTileLayer = L.Layer.extend({
 				this._selectionContentRequest = setTimeout(L.bind(function () {
 					app.socket.sendMessage('gettextselection mimetype=text/html,text/plain;charset=utf-8');}, this), 100);
 			}
+
+			if (app.socket.inputEventType === 'mouse')
```

Comment:
- check if context toolbar exists, we can have platforms / UI that doesn't support it
- what about tablet and touch? (just keep in mind, can be done later)

Reply:
Also: not a fan of distributed code and storing something in unrelated class - socket ("god object")
Can we put `inputEventType` as internal variable inside ContextToolbar (only user of it) and set by a setter - `contextToolbar.setLastEventType(mouse);`

Then here you call only:
```
const contextToolbar = this._map.contextToolbar;
if (contextToolbar) {
    var topLeftTwips = new L.Point(rectArray[0].min.x, rectArray[0].min.y);
    var position = this._twipsToPixels(topLeftTwips);
    contextToolbar.showContextToolbar(position); <-- this will check inside if allowed to show and calculate position offset as only global objects are needed, also canvas sections are TS already
}
```